From d9cca00ee16821294e4f94bc447fc1c6967aa66e Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Sun, 09 May 2021 23:02:14 +0300
Subject: [PATCH] [SW-0] rma: rma info flash boundary update

Currently the boundary condition to check for
flash read request is with reference to the
block size of sector 0. But rma info can also
be read from sector 1.

As the rma info block sizes of both sector 0
and sector 1 is same, the update does not have
a functionality impact.

Change-Id: I8efe6cf7fc4164a0e7194c33e6973cffdae834bc
---

diff --git a/gaudi/rma_info.c b/gaudi/rma_info.c
index 00f3962..b2104f4 100644
--- a/gaudi/rma_info.c
+++ b/gaudi/rma_info.c
@@ -75,9 +75,6 @@
 	u32 flash_offset;
 	struct flash_header header0, header1;
 
-	if (size + offset > flash_mem_context.flash_size[0])
-		return 1;
-
 	/* on every read we first choose a valid sector to read from */
 	spi_copy(&header0, flash_mem_context.base[0],
 				sizeof(struct flash_header), "RMA header0");
@@ -107,6 +104,9 @@
 		return 0;
 	}
 
+	if (size + offset > flash_mem_context.flash_size[flash_select])
+		return 1;
+
 	flash_offset = flash_mem_context.base[flash_select] + offset
 						+ sizeof(struct flash_header);
 	spi_copy(data, flash_offset, size, "Flash Read");
